### PR TITLE
Improved handling of disconnect / multiple instances and non-listed methods

### PR DESCRIPTION
**Pull request by ajragusa**
Hi 

We are trying to use the Net::NetConf module for our application, but ran into several issues.  We resolved these issues, and they should all be in this pull request.

Specifically there were some issues with handling / detecting loss of connection to a device, as well as being able to call methods not listed in the perl modules.

Thanks

### CODE REVIEW

**Comment by ydnath** [comment]
@ajragusa @jonstout Why change the component name and the GitHub URL in the spec and Makefiles?

File: lib/Net/Netconf/Device.pm [hunk 1]
```diff
@@ -10,6 +10,7 @@ use XML::LibXML::SAX;
 use XML::LibXML::SAX::Parser;
 use Net::Netconf::SAXHandler;
 use File::Basename;
+use Data::Dumper;
```
Comment (ydnath) [inline lib/Net/Netconf/Device.pm]: Is this Dumper used?

**Comment by jonstout** [comment]
@ydnath Apologies; I didn't realize we had a pull request open from our master branch. I'll clean our imports up a bit.

For the spec file I'll set the url to this repo, but for the name it's [common practice](https://fedoraproject.org/wiki/Packaging:Naming?rd=Packaging:NamingGuidelines#perl_modules) to prefix rpm perl modules with `perl-`; It'd like to keep the updated name if possible.

**Comment by ydnath** [comment]
@jonstout There are a few releases of this module on CPAN. Changing the module name now will disturb that. At least that is my understanding.

**Comment by ajragusa** [comment]
@ydnath I think we fixed up most of the issues you had mentioned.  Thanks

**Comment by ydnath** [comment]
@ajragusa @jonstout Thanks for the pull request. Did a quick review and it looked fine. I will merge in two days. Will release to CPAN after a week of soaking.

**Pull request merged**